mintapi/api.py

```diff
@@ -459,6 +472,87 @@ def __x_months_ago(self, months=2):
+        start_date: Optional[str] = None,
+        end_date: Optional[str] = None,
+        category_ids: List[str] = None,
+        tag_ids: List[str] = None,
+        descriptions: List[str] = None,
+        match_all_filters: bool = True,
+        limit: int = 1000,
+        offset: int = 0,
+    ) -> List[Dict]:
+        """
+        Public accessor for trend data. Internally constructs a trend api payload
+
+
+        Parameters
+        ----------
+        report_type : ReportView.Options
+            type of report to generate. must use one of enum values
+        date_filter : DateFilter.Options
+            date window. must use predefined enum windows or pass a start and end date with CUSTOM enum
+        start_date : Optional[str], optional
+            optional start date (YYYY-mm-dd) if using enum CUSTOM, by default None
+        end_date : Optional[str], optional
+            optional end date (YYYY-mm-dd) if using enum CUSTOM, by default None
+        category_ids : List[str], optional
+            optional list of category ids to filter by, by default None
+        tag_ids : List[str], optional
+            optional list of tag ids to filter by, by default None
+        descriptions : List[str], optional
+            optional list of descriptions (ui labeled merchants) to filter by, by default None
+        match_all_filters : bool, optional
+            whether to match all (True) supplied filters or any (False), by default True
+        limit : int, optional
+            page size, by default 1000
+        offset : int, optional
+            offset pagination for next pages, by default 0
+
+        Returns
+        -------
+        List[Dict]
+            returns a list of trend results (each dict)
+        """
+        search_clauses = []
+        if category_ids:
+            for category_id in category_ids:
+                search_clauses.append(
+                    CategoryMatchFilter(
+                        category_id=category_id, include_child_categories=True
+                    )
+                )
+        if tag_ids:
+            for tag_id in tag_ids:
+                search_clauses.append(TagMatchFilter(tag_id=tag_id))
+        if descriptions:
+            for description in descriptions:
+                search_clauses.append(DescriptionMatchFilter(description=description))
+
+        payload = TrendRequest(
+            report_view=ReportView(report_type=report_type),
+            date_filter=DateFilter(
+                date_filter=date_filter, start_date=start_date, end_date=end_date
+            ),
+            search_filters=SearchFilter(
+                match_all_filters=search_clauses if match_all_filters else [],
+                match_any_filters=search_clauses if not match_all_filters else [],
+            ),
+            limit=limit,
+            offset=offset,
+        )
+        return self._get_trend_data(payload=payload)
+
+    def _get_trend_data(self, payload: TrendRequest):
```

Comment:
Is there any way we can make a variation of `__call_mint_endpoint` but for POST requests?  I believe there are some other instances where we may need to send a POST.